libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -472,46 +461,18 @@ private void updateProgressChangeListener() {
 
   @Nullable
   private MapRouteProgressChangeListener buildMapRouteProgressChangeListener() {
-    shutdownVanishingRouteLineAnimator();
-    if (vanishRouteLineEnabled) {
-      vanishingRouteLineAnimator = new RouteLineValueAnimator();
-    } else {
-      vanishingRouteLineAnimator = null;
-    }
-
-    float vanishingPoint = (this.mapRouteProgressChangeListener != null)
-            ? this.mapRouteProgressChangeListener.getPercentDistanceTraveled() : 0f;
-    MapRouteProgressChangeListener newListener = new MapRouteProgressChangeListener(
-            this.routeLine,
-            routeArrow,
-            vanishingRouteLineAnimator
-    );
-    newListener.updatePercentDistanceTraveled(vanishingPoint);
-    return newListener;
-  }
-
-  /**
-   * Returns the percentage of the distance traveled that was last calculated. This is only
-   * calculated if the vanishing route line feature is enabled.
-   *
-   * @return the value calculated during the last progress update event or 0 if not enabled.
-   */
-  public float getPercentDistanceTraveled() {
-    return mapRouteProgressChangeListener.getPercentDistanceTraveled();
+    return new MapRouteProgressChangeListener(this.routeLine, routeArrow);
   }
 
   /**
-   * Can be used to manually update the percentage of route traveled.
+   * Updates the appearance of the route line representing the part of the route that has
+   * been traveled.  The line appearance will be altered from the origin point to the point
+   * indicated by the @param point
    *
-   * This is also invoked automatically when the vanishing route line feature is enabled and
-   * a new route progress update is delivered.
-   * @see #addProgressChangeListener(MapboxNavigation, boolean)
-   * @see #onNewRouteProgress(RouteProgress)
+   * @param point the end point of the traveled section of the route line.
    */
-  public void updateRouteLineWithDistanceTraveled(float distanceTraveled) {
-    routeLine.hideCasingLineAtOffset(distanceTraveled);
-    routeLine.hideRouteLineAtOffset(distanceTraveled);
-    mapRouteProgressChangeListener.updatePercentDistanceTraveled(distanceTraveled);
+  public void updateTraveledRouteLine(Point point) {
```

Comment:
Same as above - semver major change.

Reply:
I can remove this. I just remembered that when I did the original implementation I thought the developer would have to add the animator callback to the location component and then forward those points to the NavigationMapRoute. Now I remember you pointed out that NavigationMapRoute can get the locationComponent from its reference to MapboxMap.

Reply:
I reverted the mistakes in my original rebase with master. I don't think I have any breaking semver changes any longer.